### DISCUSSION TITLE
Substation transfer not maintaining new line delimited json

### DISCUSSION THREAD

**Discussion by viraj-lunani** [brexhq/substation#185] [Q&A] [answered]
I am using substation for an enrichment, however before creating the enrichment, we set up a basic config that just sends the logs to s3.
```
local sub = import "../../substation.libsonnet";
{
  concurrency: 2,
  transforms: [
      sub.tf.send.aws.s3(settings={
        bucket_name: "<bucket>",
        file_path: {
          prefix: "logs/gsuite/alerts/",
          uuid: true
        }
      }),
    ]
}
```
The logs come in as .json.gz and are writting to the bucket but when they are written to the bucket they are no longer newline delimited. How can I resolve this?

**Comment by jshlbrd** [comment, marked as answer]
@viraj-lunani I think [this example](https://github.com/brexhq/substation/blob/main/examples/config/transform/send/aux_transforms/config.jsonnet#L23) will help -- you can use the `auxiliary_transforms` setting to apply a series of transforms to the data after it is batched but before it is written to S3. We have [a pattern for JSON Lines](https://github.com/brexhq/substation/blob/main/build/config/substation.libsonnet#L1319) that you can add (also seen in the example).
